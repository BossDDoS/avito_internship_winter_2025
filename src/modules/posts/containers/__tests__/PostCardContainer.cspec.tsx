import {
  PostCardContainer,
  PostCardContainerProps,
} from '../PostCardContainter';
import { render, screen } from '@testing-library/react';

describe('PostCardContainer component', () => {
  test('Should render correct', () => {
    const props: PostCardContainerProps = {
      post: {
        id: '1',
        name: 'Toyota Camry',
        description: 'Надежный автомобиль',
        location: 'Москва',
        type: 'Авто',
        brand: 'Toyota',
        model: 'Camry',
        year: 2020,
        mileage: 15000,
      },
    };
    render(<PostCardContainer {...props} />);

    const name = screen.getByText(/Toyota Camry/);
    const description = screen.getByText(/Надежный автомобиль/);
    const location = screen.getByText(/Москва/);
    const type = screen.getByText(/Авто/);
    const brand = screen.getByText(/Toyota/);
    const model = screen.getByText(/Camry/);
    const year = screen.getByText(/2020/);
    const mileage = screen.getByText(/15000/);

    expect(name).toBeDefined();
    expect(description).toBeDefined();
    expect(location).toBeDefined();
    expect(type).toBeDefined();
    expect(brand).toBeDefined();
    expect(model).toBeDefined();
    expect(year).toBeDefined();
    expect(mileage).toBeDefined();
  });

  test('To match snapshot', () => {
    const props: PostCardContainerProps = {
      post: {
        id: '1',
        name: 'Toyota Camry',
        description: 'Надежный автомобиль',
        location: 'Москва',
        type: 'Авто',
        brand: 'Toyota',
        model: 'Camry',
        year: 2020,
        mileage: 15000,
      },
    };

    const { container } = render(<PostCardContainer {...props} />);

    expect(container).toMatchSnapshot();
  });
});
