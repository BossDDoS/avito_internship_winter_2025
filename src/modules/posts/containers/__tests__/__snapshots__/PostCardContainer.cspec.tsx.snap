// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostCardContainer component To match snapshot 1`] = `
<div>
  <div
    class="ant-card ant-card-bordered sc-FEMpB cuAMPP css-dev-only-do-not-override-1yacf91"
  >
    <div
      class="ant-card-head"
    >
      <div
        class="ant-card-head-wrapper"
      >
        <div
          class="ant-card-head-title"
        >
          Toyota Camry
        </div>
      </div>
    </div>
    <div
      class="ant-card-body"
    >
      <p>
        Надежный автомобиль
      </p>
      <p>
        <strong>
          Местоположение:
        </strong>
         
        Москва
      </p>
      <p>
        <strong>
          Тип:
        </strong>
         
        Авто
      </p>
      <p>
        <strong>
          Марка:
        </strong>
         
        Toyota
      </p>
      <p>
        <strong>
          Модель:
        </strong>
         
        Camry
      </p>
      <p>
        <strong>
          Год выпуска:
        </strong>
         
        2020
      </p>
      <p>
        <strong>
          Пробег:
        </strong>
         
        15000
         км
      </p>
    </div>
  </div>
</div>
`;
